connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/model/request/provider/GoogleVertexAiProviderConfiguration.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.agenticai.aiagent.model.request.provider;
+
+import static io.camunda.connector.agenticai.aiagent.model.request.provider.GoogleVertexAiProviderConfiguration.GOOGLE_VERTEX_AI_ID;
+
+import io.camunda.connector.generator.dsl.Property;
+import io.camunda.connector.generator.java.annotation.TemplateProperty;
+import io.camunda.connector.generator.java.annotation.TemplateSubType;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+
+@TemplateSubType(id = GOOGLE_VERTEX_AI_ID, label = "Google Vertex AI")
+public record GoogleVertexAiProviderConfiguration(
+    @Valid @NotNull GoogleVertexAiConnection googleVertexAi) implements ProviderConfiguration {
+
+  @TemplateProperty(ignore = true)
+  public static final String GOOGLE_VERTEX_AI_ID = "google-vertex-ai";
+
+  public record GoogleVertexAiConnection(
+      @NotBlank
+          @TemplateProperty(
+              group = "provider",
+              label = "Project ID",
+              description = "Specify Google Cloud project ID",
+              type = TemplateProperty.PropertyType.String,
+              feel = Property.FeelMode.optional,
+              constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+          String projectId,
+      @NotBlank
+          @TemplateProperty(
+              group = "provider",
+              label = "Location",
+              description = "Specify the region where AI inference should take place",
+              type = TemplateProperty.PropertyType.String,
+              feel = Property.FeelMode.optional,
+              constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+          String location,
+      @Valid @NotNull GoogleVertexAiProviderConfiguration.GoogleVertexAiModel model) {}
+
+  public record GoogleVertexAiModel(
+      @NotBlank
+          @TemplateProperty(
+              group = "model",
+              label = "Model",
+              description =
+                  "Specify the model ID. Details in the <a href=\"https://cloud.google.com/vertex-ai/docs/generative-ai/models\" target=\"_blank\">documentation</a>.",
+              type = TemplateProperty.PropertyType.String,
+              feel = Property.FeelMode.optional,
+              constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+          String model,
+      @Valid GoogleVertexAiModelParameters parameters) {
+
+    public record GoogleVertexAiModelParameters(
+        @Min(0)
+            @TemplateProperty(
+                group = "model",
+                label = "Maximum tokens",
```

Comment:
🔧 As this is called `maxOutputTokens` on the API docs, can we also call it "Maximum output tokens" here? We should try to align the parameters as much to the provider specific naming as possible to avoid ambiguity.

I know we have a similar construct for Bedrock where we call it `maxTokens` but call the `maxOutputTokens` builder, but there it's again called `maxTokens` on the API.